plugins/techdocs-backend/src/techdocs/stages/prepare/helpers.ts

```diff
@@ -93,3 +93,44 @@ export const checkoutGitRepository = async (
 
   return repositoryTmpPath;
 };
+
+// Could be merged with checkoutGitRepository
+export const checkoutGithubRepository = async (
+  repoUrl: string,
+): Promise<string> => {
+  const parsedGitLocation = parseGitUrl(repoUrl);
+
+  // Should propably not be hardcoded names of env variables, but seems too hard to access config down here
+  const user = process.env.GITHUB_PRIVATE_TOKEN_USER || '';
+  const token = process.env.GITHUB_PRIVATE_TOKEN || '';
+
+  const repositoryTmpPath = path.join(
+    // fs.realpathSync fixes a problem with macOS returning a path that is a symlink
+    fs.realpathSync(os.tmpdir()),
+    'backstage-repo',
+    parsedGitLocation.source,
+    parsedGitLocation.owner,
+    parsedGitLocation.name,
+    parsedGitLocation.ref,
+  );
+
+  if (fs.existsSync(repositoryTmpPath)) {
+    const repository = await Repository.open(repositoryTmpPath);
+    await repository.mergeBranches(
+      parsedGitLocation.ref,
```

Comment:
It's changed how we get this ref so that should probably be updated here as well. Problem was that `parsedGitLocation.ref` doesn't always contain a branch name

Reply:
I was also just investigating why I, in some cases, was getting a `Error: could not use 'origin/' as valid reference name` error. How are you doing this now @hooloovooo?

Reply:
You can find how its changed in this PR https://github.com/spotify/backstage/pull/2242/files or rebase on top of master and you will get the changes in this PR as well but for "checkoutGitRepository".

Reply:
Thank you! I only changed `checkoutGithubRepository` - assuming `checkoutGitRepository` wil be updated when merging :) https://github.com/spotify/backstage/pull/2247/commits/7ce7da6c71d64378a5ece99e940def136f1c031b